### ISSUE TITLE
SSH Authentication is always failing

### ISSUE THREAD

**Issue by Huckler** [godotengine/godot-git-plugin#99]
I keep receiving this error when using SSH for VCS:

"GitAPI: Performing push to origin godot-git-plugin\src\git_api.cpp:576 - GitAPI: Could not connect to remote "origin". Are your credentials correct? Try using a PAT token (in case you are using Github) as your password. Error -16: Failed to retrieve list of SSH authentication methods: Failed getting response"

It hangs for about 30s-60s before giving the message above.

Remotes (I swapped from https to SSH):

origin git@github.com:Huckler/Game.git (fetch)
origin git@github.com:Huckler/Game.git (push)

SSH keys verified functional with VSCode as well as git client and other VCS-integrated apps (Terraform Enterprise).

Occurs both with passphrase and without passphrase on the keys. I tried with keys generated through ssh-keygen on Windows, ssh-keygen on Linux, and Putty on Windows.

OS is Windows 11 21H2. Plugin version v2.0.0, Godot version 3.5b1

**Comment by ChronicallySerious** [comment]
I am not able to reproduce this error on my Windows 10, with Godot 3.5 beta1 and plugin v2.0.0.

I tried looking more on this specific error and it seems it is sometimes fixed by:
- Running `ssh-keygen -e -f ~/.ssh/id_rsa -m PEM > ~/.ssh/id_rsa.pem` to [convert the key to a valid PEM format](https://github.com/libgit2/pygit2/issues/1013#issuecomment-679200156)
- Switching private and public keys if accidentally they got switched at some point in the past

**Comment by Huckler** [comment]
Found that same thread earlier, haha. Didn't work there either. I also saw issues around generating the key with Powershell terminal vs Command Prompt - tried both, same issue.

I'll just remove the plugin and use VSCode and will try the plugin again in later versions

**Comment by ChronicallySerious** [comment]
@Huckler Hope you find this plugin useful later on. 

As for now, I would prefer to keep this issue open so that it is visible to newcomers and me as a reminder that this was a problem. You may use the unsubscribe button at the top right to stop receiving notifications/emails from the conversations under this thread.

**Comment by InkRobert** [comment]
I'm having the same problem, maybe I didn't set it up correctly a project in github, maybe you could make a training video in which you would explain to beginners how to work with it correctly.

**Comment by ChronicallySerious** [comment]
A training video would be nice, but it has the risk of containing outdated information later on because the workflow may change later on. Instead of that, I created a wiki page to set up the network features of this plugin at - https://github.com/godotengine/godot-git-plugin/wiki/Git-plugin-v2#faqs

**Comment by InkRobert** [comment]
Possibly, but I'm still getting this error, and maybe I didn't properly set up the project in git, any notes?

**Comment by ChronicallySerious** [comment]
@InkRobert Could you check if your `.ssh` folder has the right permissions? In Windows, you might be able to solve this by running Godot with admin rights. On Linux, you may check the permissions by running `ls -la ~/.ssh`

**Comment by InkRobert** [comment]
I created a new project and everything works. But on my main project, which was created on 3.2, an error appears
godot-git-plugin\src\git_api.cpp:586 - GitAPI: Failed to push. Error -1: failed to write chunk header: ￐ﾡ￐ﾵ￑ﾀ￐ﾲ￐ﾵ￑ﾀ ￐ﾲ￐ﾵ￑ﾀ￐ﾽ￑ﾃ￐ﾻ ￐ﾽ￐ﾵ￐ﾴ￐ﾾ￐﾿￑ﾃ￑ﾁ￑ﾂ￐ﾸ￐ﾼ￑ﾋ￐ﾹ ￐ﾸ￐ﾻ￐ﾸ ￐ﾽ￐ﾵ￑ﾀ￐ﾰ￑ﾁ￐﾿￐ﾾ￐ﾷ￐ﾽ￐ﾰ￐ﾽ￐ﾽ￑ﾋ￐ﾹ ￐ﾾ￑ﾂ￐ﾲ￐ﾵ￑ﾂ
:

**Comment by ChronicallySerious** [comment]
@InkRobert The push/pull/fetch features that exist in the v2 of this plugin only work with Godot 3.5. For Godot 3.2 to 3.4, you need to use the v1, which unfortunately has a very limited feature set.

**Comment by ChronicallySerious** [comment]
Also, is the SSH authentication now working in your new project?

**Comment by InkRobert** [comment]
Yes, I'm using PEM on a project and it works on a new project, but the main project was created on 3.2, but I ran it on 3.5 beta 1, and it doesn't work!

**Comment by ChronicallySerious** [comment]
That looks like a separate issue then. Could you please open a new issue for it?

**Comment by Calinou** [comment]
For reference, the new issue opened by @InkRobert is https://github.com/godotengine/godot-git-plugin/issues/102.

**Comment by naim** [comment]
FWIW, I get this error when I open the editor up and try to fetch/push/pull without remembering to open the settings and put in the passphrase for my SSH key again. Adding the passphrase and trying again resolves the issue.

It would be nice if the plugin was able to use the credential helper set in my `.gitconfig` file:
```
[credential "https://github.com"]
    helper = !/opt/homebrew/bin/gh auth git-credential
```

**Comment by ChronicallySerious** [comment]
Yes, the Godot should technically reach into the keychain or the cred helper to retrieve that. However, there needs to be some more discussion on this because any new changes we do, we also need to ensure we don't put barriers to implementing this proposal - https://github.com/godotengine/godot-proposals/issues/4369.

I am planning to implement it for Godot 4.0 after this plugin is ported to GDExtension

**Comment by naim** [comment]
Sounds good. Thanks for the comment!
